model/events.go

```diff
@@ -4,21 +4,31 @@ import (
 	"encoding/json"
 	"fmt"
 	"strconv"
+
+	"github.com/RedHatInsights/sources-api-go/util"
 )
 
 type EventModelDao interface {
-	BulkMessage(id *int64) (map[string]interface{}, error)
-	FetchAndUpdateBy(id *int64, updateAttributes map[string]interface{}) error
-	ToEventJSON(id *int64) ([]byte, error)
+	BulkMessage(resource util.Resource) (map[string]interface{}, error)
+	FetchAndUpdateBy(resource util.Resource, updateAttributes map[string]interface{}) error
+	ToEventJSON(resource util.Resource) ([]byte, error)
 }
 
-func UpdateMessage(eventObject *EventModelDao, resourceID int64, resourceType string, attributes []string) ([]byte, error) {
+func UpdateMessage(eventObject *EventModelDao, resource util.Resource, attributes []string) ([]byte, error) {
```

Comment:
We can make this `EventModelDao` not a pointer anymore since it is an interface (unless we're going to do that in a followup later)

Reply:
I tried but I need to play more with *EventModelDao -> EventModelDao, so let's do it in follow up PR.